proto/redpanda/api/dataplane/v1/mcp.proto

```diff
@@ -0,0 +1,664 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "google/api/resource.proto";
+import "google/protobuf/field_mask.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+import "redpanda/api/auth/v1/authorization.proto";
+
+// Defines the MCP server resource.
+message MCPServer {
+  option (google.api.resource) = {
+    type: "redpanda.api.dataplane.v1/MCPServer"
+    singular: "mcp_server"
+    plural: "mcp_servers"
+  };
+  // MCP Server ID.
+  string id = 1 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_/]+$"
+  ];
+
+  // User-friendly MCP server name.
+  string display_name = 2 [
+    (google.api.field_behavior) = REQUIRED,
+    (buf.validate.field).required = true,
+    (buf.validate.field).string.pattern = "^[A-Za-z0-9-_ /]+$",
+    (buf.validate.field).string.min_len = 3,
+    (buf.validate.field).string.max_len = 128
+  ];
+
+  // Optional MCP server description.
+  string description = 3 [(buf.validate.field).string.max_len = 256];
+
+  // A configuration file for the MCP server.
+  message Configuration {
+    // The path to the configuration file, this path must be relative and also valid in to layout in a filesystem,
+    // when considering many paths in a full configuration.
+    string path = 1 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true,
+      // Force a relative path pattern
+      (buf.validate.field).string.pattern = "^[A-Za-z0-9-_][A-Za-z0-9-_ /]+$",
+      (buf.validate.field).string.min_len = 3,
+      (buf.validate.field).string.max_len = 128
+    ];
+    // The contents of the configuration file in YAML format.
+    string contents = 2 [
+      (google.api.field_behavior) = REQUIRED,
+      (buf.validate.field).required = true,
+      (buf.validate.field).string.min_len = 1
+    ];
+  }
+
+  // All the configuration files for the MCP server.
+  repeated Configuration config_files = 4 [
```

Comment:
i wonder if we want to selectively update files.

if yes (i think so?), using a map<string, Configuration> will be easier with fieldmask. Updating specific indices (or even based on a field name) via fieldmask is not possible, however with maps it's quite easy, because we can use the map key in the fieldmask to update a specific map entry.

Reply:
I've been burned from using maps in the past (ie you want to change your keys, etc), but you're right it's cleaner. Will switch